src/dataset-builder/DatasetBuilderUtils.ts

```diff
@@ -114,54 +69,60 @@ export interface Cohort extends DatasetBuilderType {
   criteriaGroups: CriteriaGroup[];
 }
 
-export interface DatasetBuilderType {
-  name: string;
-}
-
 export type DatasetBuilderValue = DatasetBuilderType;
 
 export type ValueSet = {
   domain: string;
   values: DatasetBuilderValue[];
 };
 
-export interface GetConceptsResponse {
-  result: Concept[];
-}
-
-export interface SnapshotBuilderParentConcept {
-  parentId: number;
-  children: Concept[];
-}
-
-export interface GetConceptHierarchyResponse {
-  readonly result: SnapshotBuilderParentConcept[];
-}
-
-export interface SearchConceptsResponse {
-  result: Concept[];
-}
-
 export type DatasetRequest = {
   cohorts: Cohort[];
-  conceptSets: ConceptSet[];
+  conceptSets: SnapshotBuilderDatasetConceptSet[];
   valueSets: ValueSet[];
 };
 
-export type DatasetAccessRequest = {
+export type SnapshotAccessRequest = {
   name: string;
   researchPurposeStatement: string;
   datasetRequest: DatasetRequest;
 };
 
-export const convertValueSet = (valueSet: ValueSet): ValueSetApi => {
+export type SnapshotBuilderCountRequest = {
+  cohorts: Cohort[];
+};
+
+export interface DomainOption extends SnapshotBuilderOption {
+  kind: 'domain';
+  root: SnapshotBuilderConcept;
+  conceptCount?: number;
+  participantCount?: number;
```

Comment:
If these aren't non-optional in the API contract, they should be. At one point they were optional because we hadn't computed them yet. Can you make a ticket for the work to fix the API contract and then make them non-optional in the UI code?

Reply:
I'll write a ticket! sounds good!